wallet.cc

```diff
@@ -262,10 +326,119 @@ Wallet::~Wallet()
+    }
+    res = sqlite3_step(insert);
+    if (res != SQLITE_DONE) {
+        std::cerr << "Running SQL statement [\"" << sqlite3_expanded_sql(insert) << "\"] returned unexpected status code: " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        sqlite3_finalize(insert);
+        return false;
+    }
+    int secret_id = sqlite3_last_insert_rowid(m_db);
+    sqlite3_finalize(insert); insert = nullptr;
+
+    // And then write the output record
+    const std::string stmt2 = "INSERT INTO 'output' ('timestamp', 'hash', 'secret_id', 'amount', 'spent') VALUES(?, ?, ?, ?, ?);";
+    PublicWebcash pk(sk);
+    res = sqlite3_prepare_v2(m_db, stmt2.c_str(), stmt2.size(), &insert, nullptr);
+    if (res != SQLITE_OK) {
+        std::cerr << "Unable to prepare SQL statement [\"" << stmt2 << "\"]: " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        return false;
+    }
+    res = sqlite3_bind_int64(insert, 1, now);
+    if (res != SQLITE_OK) {
+        std::cerr << "Unable to bind 'timestamp' in SQL statement [\"" << stmt2 << "\"] to " << to_string(now) << ": " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        sqlite3_finalize(insert);
+        return false;
+    }
+    res = sqlite3_bind_blob(insert, 2, pk.pk.begin(), 32, SQLITE_STATIC);
+    if (res != SQLITE_OK) {
+        std::cerr << "Unable to bind 'hash' in SQL statement [\"" << stmt2 << "\"] to x'" << absl::BytesToHexString(absl::string_view((const char*)pk.pk.begin(), 32)) << "': " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        sqlite3_finalize(insert);
+        return false;
+    }
+    res = sqlite3_bind_int(insert, 3, secret_id);
+    if (res != SQLITE_OK) {
+        std::cerr << "Unable to bind 'secret_id' in SQL statement [\"" << stmt2 << "\"] to " << to_string(secret_id) << ": " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        sqlite3_finalize(insert);
+        return false;
+    }
+    res = sqlite3_bind_int64(insert, 4, pk.amount.i64);
+    if (res != SQLITE_OK) {
+        std::cerr << "Unable to bind 'amount' in SQL statement [\"" << stmt2 << "\"] to " << to_string(pk.amount.i64) << ": " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        sqlite3_finalize(insert);
+        return false;
+    }
+    res = sqlite3_bind_int(insert, 5, !!false);
+    if (res != SQLITE_OK) {
+        std::cerr << "Unable to bind 'spent' in SQL statement [\"" << stmt2 << "\"] to FALSE: " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        sqlite3_finalize(insert);
+        return false;
+    }
+    res = sqlite3_step(insert);
+    if (res != SQLITE_DONE) {
+        std::cerr << "Running SQL statement [\"" << sqlite3_expanded_sql(insert) << "\"] returned unexpected status code: " << sqlite3_errstr(res) << " (" << to_string(res) << ")" << std::endl;
+        result = false;
+    }
+    sqlite3_finalize(insert);
```

Comment:
does this do a commit in the sqlite session?

Reply:
Honestly I don't know. I'll look into how the sqlite C api handles session state. The data is being committed, but maybe there is still a further API call I should be using.

Reply:
Yes, if this isn't in a transaction (which it is not), then it commits.